animation_nodes/sockets/sound.py

```diff
@@ -1,46 +1,21 @@
+import os
 import bpy
 from bpy.props import *
 from .. events import propertyChanged
-from .. tree_info import nodeOfTypeExists
 from .. base_types import AnimationNodeSocket
-from .. algorithms.hashing import strToEnumItemID
-from .. data_structures import AverageSound, SpectrumSound
-from .. utils.nodes import newNodeAtCursor, invokeTranslation
+from .. data_structures import Sound, SoundSequence
+from .. utils.sequence_editor import getOrCreateSequencer, getEmptyChannel
 
-typeFilterItems = [
-    ("ALL", "All", "", "NONE", 0),
-    ("AVERAGE", "Average", "", "NONE", 1),
-    ("SPECTRUM", "Spectrum", "", "NONE", 2)]
-
-def getBakeDataItems(self, context):
+def getSoundSequenceItems(self, context):
     items = []
-    sequences = getattr(self.nodeTree.scene.sequence_editor, "sequences", [])
-    for index, sequence in enumerate(sequences):
-        if sequence.type != "SOUND": continue
-
-        if self.typeFilter in {"ALL", "AVERAGE"}:
-            items.extend(iterAverageItems(index, sequence))
-
-        if self.typeFilter in {"ALL", "SPECTRUM"}:
-            items.extend(iterSpectrumItems(index, sequence))
-
-    items.append(("NONE", "None", "", "NONE", 0))
-    return items
-
-def iterAverageItems(sequenceIndex, sequence):
-    for bakeIndex, data in enumerate(sequence.sound.bakedData.average):
-        yield ("AVERAGE_{}_{}".format(sequenceIndex, bakeIndex),
-               "#{} - {} - Average".format(bakeIndex, sequence.name),
-               "Low: {}  High: {}  Attack: {:.3f}  Release: {:.3f}".format(
-                   data.low, data.high, data.attack, data.release),
-               strToEnumItemID(data.identifier))
-
-def iterSpectrumItems(sequenceIndex, sequence):
-    for bakeIndex, data in enumerate(sequence.sound.bakedData.spectrum):
-        yield ("SPECTRUM_{}_{}".format(sequenceIndex, bakeIndex),
-               "#{} - {} - Spectrum".format(bakeIndex, sequence.name),
-               "Attack: {:.3f}  Release: {:.3f}".format(data.attack, data.release),
-               strToEnumItemID(data.identifier))
+    i = 0
+    for scene in bpy.data.scenes:
+        if scene.sequence_editor is not None:
+            for strip in scene.sequence_editor.sequences_all:
+                if strip.type == "SOUND":
+                    items.append((strip.name, strip.name, "", i))
```

Comment:
This simple approach might be problematic when sequences are removed.
When a sequence is removed, many enum items change their values -> the selected item changes without the user noticing.

Can you try that?

Reply:
I see. I tried that and this is indeed what happens. How should we fix this?

Reply:
You can use a similar approach to the one I used before: Use some kind of hash to compute the value. This is definitely not a perfect solution but worked relatively well in my experience.

Reply:
I think it works properly now. However, blender will return a warning about the missing enum ID. The warning is not fatal or anything, so we can just ignore it and return an empty sound, which is what I did.